### PR TITLE
fix(Text): show tip when truncate and observe resize

### PR DESCRIPTION
**Pull request by matheushahnn**
#### What does this PR do?
It fixes those issues:
https://github.com/grommet/grommet/issues/5842
https://github.com/grommet/grommet/issues/5841

#### Where should the reviewer start?
It can started by Text component and check the behavior in storybook.

#### How should this be manually tested?
Add a component in Text/stories/Tip.js to test the behavior
```
<Box width="huge">
  <Text truncate="tip"
    tip={{
      content: <Box background="brand">my custom tip content</Box>,
    }}
  >
    {alphabet}
  </Text>
</Box>
```
When not truncated it doesn't show the tooltip.
Resize the screen until truncate the text and hove the element, it must show the tooltip.
Refresh the page with the text truncated and not truncated.

#### What are the relevant issues?
* https://github.com/grommet/grommet/issues/5842
* https://github.com/grommet/grommet/issues/5841
#### Screenshots (if appropriate)
image
image

#### Is this change backwards compatible or is it a breaking change?
Yes

### CODE REVIEW

File: src/js/components/Text/Text.js [hunk 1]
```diff
@@ -24,19 +26,22 @@ const Text = forwardRef(
...
+        setTextTruncated(true);
       }
-    }, [children, textRef, truncate]);
+    };
+
+    useLayoutEffect(() => {
+      window.addEventListener('resize', updateTip);
+      updateTip();
+      return () => window.removeEventListener('resize', updateTip);
+    }, []);
```
Comment (jcfilben) [inline src/js/components/Text/Text.js:44]:
```suggestion
    }, [updateTip]);
```

**Review by jcfilben — COMMENTED** [review]
Just a minor comment, otherwise looks good

File: src/js/components/Text/Text.js (hunk 1 above)
Comment (matheushahnn) [reply src/js/components/Text/Text.js]: Thank you for your suggestion.
I move the updateTip inside to useLayoutEffect().

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**